examples/cloud-operations/network-dashboard/main.tf

```diff
@@ -0,0 +1,183 @@
+  limit_instances          = join(",", local.limit_instances_list)
+  limit_instances_ppg_list = tolist(var.limit_instances_ppg)
+  limit_instances_ppg      = join(",", local.limit_instances_ppg_list)
+  limit_vpc_peer_list      = tolist(var.limit_vpc_peer)
+  limit_vpc_peer           = join(",", local.limit_vpc_peer_list)
+  limit_l4_list            = tolist(var.limit_l4)
+  limit_l4                 = join(",", local.limit_l4_list)
+  limit_l7_list            = tolist(var.limit_l7)
+  limit_l7                 = join(",", local.limit_l7_list)
+  limit_l4_ppg_list        = tolist(var.limit_l4_ppg)
+  limit_l4_ppg             = join(",", local.limit_l4_ppg_list)
+  limit_l7_ppg_list        = tolist(var.limit_l7_ppg)
+  limit_l7_ppg             = join(",", local.limit_l7_ppg_list)
+}
+
+################################################
+# Monitoring project creation                  #
+################################################
+
+module "project-monitoring" {
+  source          = "github.com/GoogleCloudPlatform/cloud-foundation-fabric///modules/project"
+  name            = "monitoring"
+  parent          = "organizations/${var.organization_id}"
+  prefix          = var.prefix
+  billing_account = var.billing_account
+  services        = var.project_monitoring_services
+}
+
+################################################
+# Service account creation and IAM permissions #
+################################################
+
+module "service-account-function" {
+  source       = "github.com/GoogleCloudPlatform/cloud-foundation-fabric///modules/iam-service-account"
+  project_id   = module.project-monitoring.project_id
+  name         = "sa-dash"
+  generate_key = false
+
+  # Required IAM permissions for this service account are:
+  # 1) compute.networkViewer on projects to be monitored (I gave it at organization level for now for simplicity)
+  # 2) monitoring viewer on the projects to be monitored (I gave it at organization level for now for simplicity)
+  iam_organization_roles = {
+    "${var.organization_id}" = [
+      "roles/compute.networkViewer",
+      "roles/monitoring.viewer",
+    ]
+  }
+
+  iam_project_roles = {
+    "${module.project-monitoring.project_id}" = [
+      "roles/monitoring.metricWriter"
+    ]
+  }
+}
+
+################################################
+# Cloud Function configuration (& Scheduler)   #
+################################################
+
+# Create an app engine application (required for Cloud Scheduler)
+resource "google_app_engine_application" "scheduler_app" {
+  project = module.project-monitoring.project_id
+  # "europe-west1" is called "europe-west" and "us-central1" is "us-central" for App Engine, see https://cloud.google.com/appengine/docs/locations
+  location_id = var.region == "europe-west1" || var.region == "us-central1" ? substr(var.region, 0, length(var.region) - 1) : var.region
+}
+
+# Create a storage bucket for the Cloud Function's code
```

Comment:
nit: the `cloud-function` module does the same thing you're doing here with GCS, the zip file, etc. No need to change but it would make the code a bit simpler

Reply:
I would have to test that, do you handle the europe-west1 region being called "europe-west" by AppEngine? I know I had to include an ugly test for 2 corner cases.

Reply:
I now use the cloud-function module.